x-pack/plugin/redact/src/main/java/org/elasticsearch/xpack/redact/RedactProcessor.java

```diff
@@ -350,6 +384,12 @@ public RedactProcessor create(
             }
             Map<String, String> customPatternBank = ConfigurationUtils.readOptionalMap(TYPE, processorTag, config, "pattern_definitions");
 
+            // on-creation license check, this is exercised at processor creation time
+            if (skipIfUnlicensed == false && REDACT_PROCESSOR_FEATURE.check(licenseState) == false) {
```

Comment:
```suggestion
            if (REDACT_PROCESSOR_FEATURE.check(licenseState) == false) {
```

My understanding is that the `skip_if_unlicensed` is to prevent the redact processor failing the entire ingest pipeline once the licence expires. I'm less sure about allowing a user to optimistically create a processor with `skip_if_unlicensed` hoping it will become licensed later. 

It now strikes me that `skip_if_unlicensed` could be covered by the `ignore_failure` option that is common to all processors. To some, it may be confusing for the processor to fail a licence check if `ignore_failure: true`.

Reply:
> I'm less sure about allowing a user to optimistically create a processor with skip_if_unlicensed hoping it will become licensed later.

That's pretty much the point of this work (https://github.com/elastic/elasticsearch/issues/94882), however -- if the `elastic/integrations` want to ship a `redact` processor by default, then we need to gracefully handle the license-not-sufficient case, since it could happen on every single document at scale and is not an unexpected case.

Another consideration is that the performance of `ignore_failure` isn't actually especially good, due to the implementation details of how it happens to be implemented (note, however, that that isn't set in stone, we could 'just' improve that).

In the same space -- `ignore_missing` (which many processors including redact already have) is also unnecessary, but we special case it for similar reasons. 

-----

All that said, though, I'm sympathetic. A more single-task-focused version of this PR could drop `skip_if_unlicensed` entirely, leaving that as an enhancement for another day. I'll see what kind of feedback I can get on going that way.

Reply:
>That's pretty much the point of this work (https://github.com/elastic/elasticsearch/issues/94882)

Thanks for the background, I see the need now

> ignore_missing ... is also unnecessary, but we special case it for similar reasons.

👍  it makes sense for `ignore_failure` and `skip_if_unlicensed` as distinct events

The InferenceProcessor could implement the same `skip_if_unlicensed` option by adding logic catch and unwrap the compliance exception and skip error reporting if `skip_if_unlicensed:true`. 

The trained model licensing error is thrown here: 
https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportInternalInferModelAction.java#L125